### PR TITLE
GH-2654: Incorrect retry topic suffix

### PR DESCRIPTION
**Pull request by Wzy19930507**
Resolves https://github.com/spring-projects/spring-kafka/issues/2654

* fix incorrect retry topic suffix when attempts is 2

* optimization method `isDelayWithReusedTopic`

* add retry topic integration tests when reuse retry topic

fix java doc in RetryTopic

* fix javadoc in `@RetryTopic`

* fix javadoc in `RetryTopicConfigurationProvider`

### CODE REVIEW

**Review by artembilan — APPROVED** [review]
This is OK with me, but we still need @garyrussell 's approval.

And I guess since we have a workaround for older versions, we are not going to back-port the fix since it feels a bit as a breaking change.

Thanks

**Comment by Wzy19930507** [comment]
Thanks for your review 💯

**Review by garyrussell — CHANGES_REQUESTED** [review]
LGTM, but we should add a note to the `whats-new.adoc` about the behavior change - users will somehow need to migrate any messages in the "old" retry topic.

On the other hand, maybe the new behavior needs to be an opt-in property, to avoid the need for such migration.

We are getting close to GA (next week); if we decide to go with the opt-in approach, it would be ok to move it to 3.1.1.

**Comment by Wzy19930507** [comment]
> On the other hand, maybe the new behavior needs to be an opt-in property, to avoid the need for such migration.

May we use this case replacement opt-in?
```
@RetryableTopic(
        backoff = Backoff(delay = 5000),
        attempts = "2",
        autoCreateTopics = "false",
        fixedDelayTopicStrategy = FixedDelayStrategy.SINGLE_TOPIC
        
    )
```

File: spring-kafka-docs/src/main/antora/modules/ROOT/pages/whats-new.adoc [hunk 1]
```diff
@@ -38,3 +38,9 @@ See xref:kafka/container-factory.adoc[Container Factory] for more information.
 You can now add a `Validator` to this deserializer; if the delegate `Deserializer` successfully deserializes the object, but that object fails validation, an exception is thrown similar to a deserialization exception occurring.
 This allows the original raw data to be passed to the error handler.
 See xref:kafka/serdes.adoc#error-handling-deserializer[Using `ErrorHandlingDeserializer`] for more information.
+
+[[x31-retryable]]
+=== Retryable Topics
+Change suffix -retry-5000 to -retry when `@RetryableTopic(backoff = @Backoff(delay = 5000), attempts = "2", fixedDelayTopicStrategy = FixedDelayStrategy.SINGLE_TOPIC)`.
+If you want to keep suffix -retry-5000, use `@RetryableTopic(backoff = @Backoff(delay = 5000), attempts = "2")`.
+See xref:retrytopic/topic-naming.adoc[Topic Naming] for more information.
```
Comment (garyrussell) [inline spring-kafka-docs/src/main/antora/modules/ROOT/pages/whats-new.adoc:46]:
```suggestion
Change suffix `-retry-5000` to `-retry` when `@RetryableTopic(backoff = @Backoff(delay = 5000), attempts = "2", fixedDelayTopicStrategy = FixedDelayStrategy.SINGLE_TOPIC)`.
If you want to keep suffix `-retry-5000`, use `@RetryableTopic(backoff = @Backoff(delay = 5000), attempts = "2")`.
See xref:retrytopic/topic-naming.adoc[Topic Naming] for more information.
```

**Review by garyrussell — APPROVED** [review]

**Comment by garyrussell** [comment]
Thanks!

Please rebase to main to address conflicts.

**Comment by Wzy19930507** [comment]
Thanks you for review : )

**Pull request merged**